### PR TITLE
[WIP] Create ts file and add tsconfig

### LINKED ISSUE #81: Refactor
- [x] Convert to typescript with one file and compile and see if the demo works
- [ ] disable `noImplicitAny`
   - [ ] For Vega-Lite properties, import the types (e.g., `FieldDef` from `'vega-lite/src/fielddef'`)
   - [ ] For `event` it's probably HTML event, google if TS has interface already.  If not, we can cheat by say `type Event = any;`  then use `event: Event`.

- [ ] split into two files `vega-tooltip.ts` and `vega-lite-tooltip.ts` the latter can depend on the former. 

- [ ] Further split files
  - [ ] `prepareFieldsData.ts` for `prepareAllFieldsData`, `prepareCustomFieldsData` 
  - [ ] Distinguish between parts that generates data and parts that modify DOM.

### PR DESCRIPTION
**Pull request by sirahd**
- Fix #81 on converting to typescript and disables `noImplicitAny`.

### CODE REVIEW

**Comment by kanitw** [comment]
Not sure why you create this PR?  Is this a WIP?  If so, please *always* mark this as WIP.

A few obvious things to do:

- [ ]  Add the old js file to npmignore and remove it from the source code. 
- will comment more inline

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    "noEmitOnError": true,
+    "noImplicitThis": true,
+    "noImplicitReturns": true,
+    "sourceMap": true,
+    "typeRoots" : ["./typings", "node_modules/@types"]
+  },
+  "include": [
+    "src/**/*.ts",
+    "test/**/*.ts",
+    "examples/**/*.ts",
```
Comment (kanitw) [inline tsconfig.json]: I don't think you have this "example" folder.   Similar for things in `exclude`.

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{
+  "compilerOptions": {
+    "target": "es5",
```
Comment (kanitw) [inline tsconfig.json]: I would add `"outDir": `./build`.   (We plan to add that for Vega-Lite pretty soon too.)

**Comment by kanitw** [comment]
- [ ] See how we config CompassQL might be helpful for you :) 

- See https://github.com/vega/compassql/compare/eedcdfaf440fc910fff727c043237f2cbba601c6...0cbbb186d37a58e35a69f7c7f6e2817b2fd4bfcd?expand=1  
- Or clone CompassQL and see the setup.

**Comment by sirahd** [comment]
For the "old js file", do you mean the one in root directory (`vg-tooltip.min.js`) or the compiled version inside `src` directory?

File: .gitignore [hunk 3]
```diff
@@ -3,4 +3,5 @@ bower_components
 node_modules
 /vg-tooltip.min.js
 /vg-tooltip.css
+src/vg-tooltip.js
```
Comment (kanitw) [inline .gitignore]: Actually you shouldn't add this to both npm and gitignore, but should simply remove it from the src folder.  (`git rm src/vg-tooltip.js`)

**Pull request merged**